frontend/catalyst/pennylane_extensions.py

```diff
@@ -242,22 +315,25 @@ def default_expand_fn(self, circuit, max_expansion=10):
         if any(isinstance(op, MidMeasureMP) for op in circuit.operations):
             raise CompileError("Must use 'measure' from Catalyst instead of PennyLane.")
 
+        decompose_to_qubit_unitary = QJITDevice._get_operations_to_convert_to_matrix(self.config)
+
+        def _decomp_to_unitary(self, *_args, **_kwargs):
+            return [qml.QubitUnitary(qml.matrix(self), wires=self.wires)]
+
         # Fallback for controlled gates that won't decompose successfully.
         # Doing so before rather than after decomposition is generally a trade-off. For low
         # numbers of qubits, a unitary gate might be faster, while for large qubit numbers prior
         # decomposition is generally faster.
         # At the moment, bypassing decomposition for controlled gates will generally have a higher
         # success rate, as complex decomposition paths can fail to trace (c.f. PL #3521, #3522).
-
-        def _decomp_controlled(self, *_args, **_kwargs):
-            return [qml.QubitUnitary(qml.matrix(self), wires=self.wires)]
-
-        with Patcher(
-            (qml.ops.Controlled, "has_decomposition", lambda self: True),
-            (qml.ops.Controlled, "decomposition", _decomp_controlled),
-            # TODO: Remove once work_wires is no longer needed for decomposition.
-            (qml.ops.MultiControlledX, "decomposition", _decomp_controlled),
-        ):
+        overriden_methods = [  # pragma: no cover
+            (qml.ops.Controlled, "has_decomposition", lambda self: True),  # pragma: no cover
+            (qml.ops.Controlled, "decomposition", _decomp_to_unitary),  # pragma: no cover
+        ]  # pragma: no cover
```

Comment:
🤣 Didn't know which one to mark?

Reply:
Removed all but one: https://github.com/PennyLaneAI/catalyst/pull/369/commits/3fb64d68ad46191e3727ed2dcc1fe5f08349030a